invenio_records_resources/services/records/service.py

```diff
@@ -211,12 +211,21 @@ def reindex(self, identity, params=None, es_preference=None, **kwargs):
         """Reindex records matching the query parameters."""
         self.require_permission(identity, 'search')
 
-        # Prepare and execute the search as scan()
+        # prepare and update query
         params = params or {}
-        search_result = self._search(
-            'reindex', identity, params, es_preference, **kwargs).scan()
+        params.update(kwargs)
```

Comment:
Not sure about moving `kwargs` here since it was the only way before to pass things to `self.search_request(...)`. I'm aware the API is a bit "vague" and has a lot of indirection, but in that case, I would take a step back and review the overall usage before changing "core" parts.

Reply:
Ok, just noticed that it used to be `self._search(...)` before (vs. `self.search_request(...)`). I think as we discussed IRL it's better to go directly for the "low-level" API :)